lib/internal/modules/esm/resolve.js

```diff
@@ -19,7 +19,28 @@ const { ERR_INPUT_TYPE_NOT_ALLOWED,
 
 const realpathCache = new SafeMap();
 
-function defaultResolve(specifier, { parentURL } = {}, defaultResolve) {
+const DEFAULT_CONDITIONS_KEY = 'node\0import';
+const DEFAULT_CONDITION_SET = new Set(['node', 'import']);
```

Comment:
I think it may be better to have this cache as a `std::set<std::string>` within the C++ code and only generate the key here. `default` is on purpose not included. I think it makes sense if loaders cannot influence it.